.github/workflows/integrations.yml

```diff
@@ -194,8 +194,49 @@ jobs:
           CMAKE_PREFIX_PATH: ${{ github.workspace }}/dist
           CMAKE_TOOLCHAIN_FILE: ${{ github.workspace }}/.conan/conan_paths.cmake
 
-      # TODO(DF): SimpleCppManager should (also) be tested as part of
-      # Python e2e tests, once we fix loading C++ plugins from Python.
+  cpp-manager-plugin:
+    name: C++ manager plugin on ${{ matrix.config.os }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - os: windows-2019
+            preamble: call "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Auxiliary\\Build\\vcvarsall.bat" x64
+            site-packages: Lib/site-packages
+            shell: cmd
+          - os: ubuntu-20.04
+            site-packages: lib/python3.9/site-packages
+            shell: bash
+          - os: macos-12
+            site-packages: lib/python3.9/site-packages
+            shell: bash
+    defaults:
+      run:
+        shell: ${{ matrix.config.shell }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Bootstrap
+        uses: ./.github/bootstrap_platform
+
+      - name: Install test dependencies
+        run: |
+          python -m pip install pytest
+
+      - name: Build/install OpenAssetIO (CMake)
+        run: >
+          ${{ matrix.config.preamble }}
+
+          cmake -S . -B build --install-prefix ${{ github.workspace }}/dist
```

Comment:
Nit: As you've got a CMakePresets.json, I'm surprised it's not being used here. The advice for presets in the Professional CMake book is that they should describe your CI 'ready to release' setup.

Reply:
I think in this particular case a preset wouldn't give us much?  Other than perhaps setting the build type.

More generally we probably could make more use of presets in various bits of CI - its a little bit mixed at the moment.

There is a counter-argument, though - by not using presets we're dogfooding the defaults, which for OpenAssetIO is intended to produce a release-ready build, as would typically be desired by consumers.

Reply:
> Oh I guess you can set the build directory with presets too.

Yes

> There is a counter-argument, though - by not using presets we're dogfooding the defaults
There isn't a need to set CMake cache variables in a preset - you can still use the defaults. It's just that

```
cmake --workflow --preset ci_Release
```

which does as much as you tie into a workflow (configure + build (+ test) (+packaging)) does it all in one command, which is very appealing for CI.


The point the documentation makes is that the presets are documenting (if used for CI) what you _expect_ to always work, so is a starting point for everyone, including new developers. If you don't go down the presets route, and there are a whole bunch of options, then you do wonder 'which ones should I set?' I find that a lot in the thirdparty libraries from upstream.

Reply:
Definitely agree - added https://github.com/OpenAssetIO/OpenAssetIO/issues/1363